modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/distribution/CacheDistributionTaskResult.java

```diff
@@ -0,0 +1,344 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.commandline.cache.distribution;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.ignite.internal.processors.cache.distributed.dht.GridDhtPartitionState;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.internal.visor.VisorDataTransferObject;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Result of CacheDistributionTask
+ */
+public class CacheDistributionTaskResult extends VisorDataTransferObject {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Job results. */
+    private List<CacheDistributionNode> nodeResultList;
+
+    /** Exceptions. */
+    private Map<UUID, Exception> exceptions;
+
+    /**
+     * @param nodeResultList Cluster infos.
+     * @param exceptions Exceptions.
+     */
+    public CacheDistributionTaskResult(List<CacheDistributionNode> nodeResultList,
+        Map<UUID, Exception> exceptions) {
+        this.nodeResultList = nodeResultList;
+        this.exceptions = exceptions;
+    }
+
+    /**
+     * For externalization only.
+     */
+    public CacheDistributionTaskResult() {
+    }
+
+    /**
+     * @return Job results.
+     */
+    public Collection<CacheDistributionNode> jobResults() {
+        return nodeResultList;
+    }
+
+    /**
+     * @return Exceptions.
+     */
+    public Map<UUID, Exception> exceptions() {
+        return exceptions;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void writeExternalData(ObjectOutput out) throws IOException {
+        U.writeCollection(out, nodeResultList);
+        U.writeMap(out, exceptions);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void readExternalData(byte protoVer, ObjectInput in
+    ) throws IOException, ClassNotFoundException {
+        nodeResultList = U.readList(in);
+        exceptions = U.readMap(in);
+    }
+
+    /** {@inheritDoc} */
+    @Override public String toString() {
+        return S.toString(CacheDistributionTaskResult.class, this);
+    }
+
+    private static class Row implements Comparable {
```

Comment:
Why this class embeddbed into  CacheDistributionTaskResult?

Reply:
This class need for sort and format result